src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -17,19 +18,38 @@
     ResultsStrategyFactory,
 )
 from sentry.uptime.detectors.ranking import _get_cluster
+from sentry.uptime.detectors.tasks import set_failed_url
 from sentry.uptime.issue_platform import create_issue_platform_occurrence
-from sentry.uptime.models import ProjectUptimeSubscription, UptimeSubscription
+from sentry.uptime.models import (
+    ProjectUptimeSubscription,
+    ProjectUptimeSubscriptionMode,
+    UptimeSubscription,
+)
+from sentry.uptime.subscriptions.subscriptions import (
+    create_uptime_subscription,
+    delete_project_uptime_subscription,
+    remove_uptime_subscription_if_unused,
+)
 from sentry.utils import metrics
 from sentry.utils.hashlib import md5_text
 
 logger = logging.getLogger(__name__)
 LAST_UPDATE_REDIS_TTL = timedelta(days=7)
+ONBOARDING_FAILURE_THRESHOLD = 3
+ONBOARDING_FAILURE_REDIS_TTL = timedelta(days=2)
```

Comment:
Shouldn't this be longer than `ONBOARDING_MONITOR_PERIOD`? If we have two failures day 1 and then 1 failure in the very last day, shouldn't that actually mean we met the threshold, but because of the expiry we won't have?

Reply:
We bump the expiry on every result, so this should be fine unless we have a really extended downtime

Reply:
Seems like we only bump it on failures?

Reply:
Good point, I'll just bump it to 3 days

Reply:
I called it out in the comment here https://github.com/getsentry/sentry/pull/74280#discussion_r1678166311

I gave it +1 day, but there's probably no reason, it can just be the same as `ONBOARDING_MONITOR_PERIOD`